src/operator/tensor/matrix_op-inl.h

```diff
@@ -11,6 +11,14 @@
 #include "../mshadow_op.h"
 #include "../elemwise_op_common.h"
 
+namespace dmlc {
+
+// Add a few patches to support signed Shape in dmlc/parameter.
+DMLC_DECLARE_TYPE_NAME(nnvm::Tuple<int>, "Shape(tuple)");
```

Comment:
see https://github.com/dmlc/mxnet/blob/master/include/mxnet/c_api.h#L561, this api returns symbol information such as `arg_type_infos`. Without declaring type name for `nnvm::Tuple<int>`, information of the reshape operator, which are required by language bindings, will be incomplete.